### ISSUE TITLE
Negated matching

### ISSUE THREAD

**Issue by freemin7** [maxitg/SetReplace#323]
### The problem
Currently rules match just when certain connection conditions are satisfied. This often leads to cancerous growth patterns. It would be helpful to be able to construct rules which don't match if a certain thing exist already. The thread: 
https://community.wolfram.com/groups/-/m/t/1946413 
gives insight how more powerful the approach is and what notation might be used. We might be missing out on a large part of the computational universe by not including this as it ins not clear that all those rules can be emulated by clever "hyper edge constructions".
   

### Possible solution
Implement negated matching inside WolframModel.

### Alternative solutions
Implement the new Model maybe called AndersenModel (after the "discoverer" who is not me) whp supports negated matching

### Additional context
I am Johann-Tobias Schäg, who posed the construct a 2d grid problem.

**Comment by maxitg** [comment]
This looks quite interesting. I don't quite have the bandwidth to work on this myself right now, but if someone can implement this without breaking changes (perhaps using an `Except` wrapper for negations), I would be happy to review/merge it and experiment with it.

**Comment by daneelsan** [comment]
Haven't read the whole post but this works now:
```wl
In[]:= SetReplace[{{1, 2, 3}, {3, 4, 5}, {1, 2, 3}, {3, 4, 5}}, {{1, 2, 3}, {3, 4, 5}, Except[{5, 6, 7}]} -> {{1, 3, 5}}]
Out[]= {{3, 4, 5}, {1, 3, 5}}
```

**Comment by maxitg** [comment]
Not really, it doesn't. This `Except` has nothing to do with `{5, 6, 7}`:
```wl
In[] := ToPatternRules[{{1, 2, 3}, {3, 4, 5}, 
   Except[{5, 6, 7}]} -> {{1, 3, 5}}]
Out[] = {{v1_, v2_, v3_}, {v3_, v4_, v5_}, Except[v6_]} :> {{v1, v3, v5}}
```

This is actually quite complicated to implement. It breaks our whole approach to multiway systems as now events depend not only on their inputs but also on the rest of the state.
